pkg/skaffold/build/ko/builder_test.go

```diff
@@ -46,128 +51,108 @@ func TestBuildOptions(t *testing.T) {
 					KoArtifact: &latestV1.KoArtifact{},
 				},
 			},
-		},
-		{
-			description: "base image",
-			artifact: latestV1.Artifact{
-				ArtifactType: latestV1.ArtifactType{
-					KoArtifact: &latestV1.KoArtifact{
-						BaseImage: "gcr.io/distroless/base:nonroot",
-					},
-				},
-				ImageName: "ko://example.com/foo",
-			},
-		},
-		{
-			description: "empty platforms",
-			artifact: latestV1.Artifact{
-				ArtifactType: latestV1.ArtifactType{
-					KoArtifact: &latestV1.KoArtifact{
-						Platforms: []string{},
-					},
-				},
-				ImageName: "ko://example.com/foo",
+			wantBo: options.BuildOptions{
+				ConcurrentBuilds: 1,
+				Trimpath:         true,
+				UserAgent:        version.UserAgentWithClient(),
 			},
 		},
 		{
-			description: "multiple platforms",
+			description: "all options",
 			artifact: latestV1.Artifact{
 				ArtifactType: latestV1.ArtifactType{
 					KoArtifact: &latestV1.KoArtifact{
+						BaseImage: "gcr.io/distroless/base:nonroot",
+						Dir:       "gomoddir",
+						Env: []string{
+							"FOO=BAR",
+							fmt.Sprintf("frob={{.%s}}", testKoBuildOptionsEnvVar),
+						},
+						Flags: []string{
+							"-v",
+							fmt.Sprintf("-flag-{{.%s}}", testKoBuildOptionsEnvVar),
+						},
+						Labels: map[string]string{
+							"foo":  "bar",
+							"frob": fmt.Sprintf("{{.%s}}", testKoBuildOptionsEnvVar),
+						},
+						Ldflags: []string{
+							"-s",
+							fmt.Sprintf("-ldflag-{{.%s}}", testKoBuildOptionsEnvVar),
+						},
+						Main:      "cmd/app",
 						Platforms: []string{"linux/amd64", "linux/arm64"},
 					},
 				},
 				ImageName: "ko://example.com/foo",
+				Workspace: "workdir",
 			},
-			wantPlatform: "linux/amd64,linux/arm64",
-		},
-		{
-			description: "workspace",
-			artifact: latestV1.Artifact{
-				ArtifactType: latestV1.ArtifactType{
-					KoArtifact: &latestV1.KoArtifact{},
-				},
-				ImageName: "ko://example.com/foo",
-				Workspace: "my-app-subdirectory",
-			},
-			wantWorkingDirectory: "my-app-subdirectory",
-		},
-		{
-			description: "source dir",
-			artifact: latestV1.Artifact{
-				ArtifactType: latestV1.ArtifactType{
-					KoArtifact: &latestV1.KoArtifact{
-						Dir: "my-go-mod-is-here",
+			envVarValue: "baz",
```

Comment:
shd we add a test case for `{{.Env.baz}}`  here?

Reply:
Added.